### ISSUE TITLE
Hidden Fields are not clearing the value

### ISSUE THREAD

**Issue by ajay1208** [ngx-formly/ngx-formly#1847]
I have hiddenexpression. which shows based on the other fields. When on condition hidden the values are not cleared. Like when the field is hidden we should clear the input value also.

Current behavior
Values are not removed

Expected behavior
Values are remain.

I have provided the stackblitz example in the below link.

https://stackblitz.com/edit/angular-fdjd5u-r9dzqb

steps
1 . click option 2 radio
2. now you should see line 2. enter some value in line 2
3. now click option 1 radio
4.  line 2 will be hidden
5. now again click option 2 radio
6. you will see the value there. it will not be cleared.

**Comment by aitboudad** [comment]
see https://github.com/ngx-formly/ngx-formly/issues/620#issuecomment-355535192

**Comment by dummdidumm** [comment]
If at all, I would make this behavior flaggable, because in my case it would be bad to have this behavior.

**Comment by yluijten** [comment]
Hi @aitboudad, any updates on this one?

**Comment by aitboudad** [comment]
@yluijten I'm focusing on other stuff right now so it may take some time to be ready (scheduled for `v6`) but I've done it for ´json-schema´ as a hidden feature but you can start using it if you want by setting `autoClear` to `true`

```ts
field['autoClear'] = true;
```

you may create a custom extension to enable it for all fields.

**Comment by aitboudad** [comment]
another simple way is to rely on `form.value` which doesn't include the hidden fields value.

**Comment by pascalbe-dev** [comment]
Using form.value works fine. For everyone, who want to also include disabled control values to the final `form.value`. You can instead use `form.getRawValue()`, which includes disabled field values as stated in [this stackoverflow question](https://stackoverflow.com/a/40148168/7870650).

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v5.10.0](https://github.com/ngx-formly/ngx-formly/releases/tag/v5.10.0) release. You can now enable clearing the hidden field value using the following config:

```patch
    FormlyModule.forRoot({
+      extras: {
+        resetFieldOnHide: true,
+      },
    }),
```

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
